File: AspNetCore_gdr2333_Front+Back/Back/Types/DataBase/Treehollow.fs
namespace Back.Types.DataBase

open System
open System.ComponentModel.DataAnnotations

type Treehollow(senderId: int64, content: string, isPublic: bool) =
    class
        [<Key>]
        member val TreehollowId = 0L with get

        member val SenderId = senderId with get
        member val Content = content with get
        member val SendTime = DateTime.Now with get
        member val Checked = false with get, set
        member val CheckSucess = false with get, set
        member val IsPublic = isPublic with get, set
    end

File: AspNetCore_gdr2333_Front+Back/Back/Types/DataBase/Treehollow.cs
using System.ComponentModel.DataAnnotations;

namespace Back.Types.DataBase;

public class Treehollow
{
    [Key]
    public long TreehollowId { get; set; }

    public long SenderId { get; set; }

    public string Content { get; set; }

    public DateTime SendTime { get; set; }

    public bool Checked { get; set; }

    public bool CheckSuccess { get; set; }

    public bool IsPublic { get; set; }

    public bool Deleted { get; set; }

    public Treehollow(long senderId, string content, bool isPublic)
    {
        SenderId = senderId;
        Content = content;
        IsPublic = isPublic;
        SendTime = DateTime.Now;
        Checked = false;
        CheckSuccess = false;
        Deleted = false;
    }

    public Treehollow()
    {
    }
}
